sdk/identity/azure-identity/src/test/java/com/azure/identity/AzureCliCredentialTest.java

```diff
@@ -148,4 +148,48 @@ public void testValidMultiTenantAuth() {
             .expectErrorMatches(e -> e instanceof ClientAuthenticationException)
             .verify();
     }
+
+    @Test
+    public void testSubscriptionAccepted() {
```

Comment:
Can you parameterize this to have a subscription with all the valid characters? Or at least change the test data to include all of them.